sdk/azidentity/managed_identity_client.go

```diff
@@ -437,7 +437,7 @@ func (c *managedIdentityClient) createCloudShellAuthRequest(ctx context.Context,
 		log.Write(EventAuthentication, "WARNING: Cloud Shell doesn't support user-assigned managed identities")
 		q := request.Raw().URL.Query()
 		if id.idKind() == miResourceID {
-			q.Add(qpResID, id.String())
+			q.Add(miResID, id.String())
```

Comment:
Most places seem to keep using the mi string, instead of msi, but the PR description is implying we should convege and only use msi.

Reply:
We need both. We should use "msi_res_id" for IMDS (and implicitly, anything resembling IMDS). For other platforms such as App Service, "mi_res_id" is correct.

Reply:
Well, this was interesting. The App Service doc was changed after we implemented this feature because a reader saw that other APIs use `msi_res_id` and concluded `mi_res_id` was a typo. It isn't: `mi_res_id` is correct and App Service ignores `msi_res_id`. I opened https://github.com/MicrosoftDocs/azure-docs/pull/121176 to fix the doc.